int32 num
bool value
---
bool success
